routes/admin.js

```diff
@@ -0,0 +1,250 @@
+     *    ...
+     *  ]
+     * }
+    */
+    method: 'GET',
+    path: '/admin/{level}/units',
+    handler: function (req, res) {
+      const level = req.params.level;
+      // check to make sure admin level is of the proper type
+      const levelCheck = ['commune', 'district', 'province'].find((l) => { return l === level; });
+      if (!levelCheck) {
+        res(Boom.badRequest('Admin level must be of type \'province\', \'district\', or \'commune'));
+      }
+      // if proper, SELECT id FROM admin_boundaries WHERE type=${level}
+      knex('admin_boundaries')
+      .where({ type: level })
+      .select('id', 'name_en')
+      .then((ids) => {
+        // format response per apiSuccessExample spec
+        const resObj = {};
+        resObj[level] = ids;
+        // serve result
+        res(resObj);
+      })
+      .catch((e) => {
+        console.log(e);
+        // return error if it occurs
+        res(Boom.wrap(e));
+      });
+    }
+  },
+  {
+    /**
+     * @api {get} /admin/units?name=substring&limit=# List Matching Units
+     * @apiGroup Admin
+     * @apiName ListMathcingUnits
+     * @apiDescription Returns list of admin unit ids that match a substring.
+     * @apiVersion 0.1.0
+     *
+     * @apiParam {String} name a string used to search for matching admin units
+     * @apiParam {Number} limit a number that when passed limits the # of return units to equal limit
+     *
+     * @apiSuccess {array} units array including list of all admin ids that match the passed substring
+     * @apiSuccess {object} units[i] unit object including unit id and name_en properties
+     * @apiSuccess {Number} units[i].id unit id
+     * @apiSuccess {string} units[i].level unit level
+     * @apiSuccess {string} units[i].name_en english name of matching unit
+     *
+     *
+     * @apiSuccessExample {JSON} Example Usage:
+     *  curl http://localhost:4000/api/admins/units?name=Th&limit=2
+     *
+     * @apiSuccessExample {JSON} Success-Response
+     * [
+     *  {
+     *    id: 2071205,
+     *    name_en: "Thanh Van",
+     *    level: "commune"
+     *  },
+     *  {
+     *    id: 2071209,
+     *    name_en: "Thanh Mai",
+     *    level: "commune"
+     *  }
+     * ]
+    */
+    method: 'GET',
+    path: '/admin/units',
+    handler: function (req, res) {
+      // parse the query string and the limit setting
+      const queryString = req.query.name.toLowerCase();
+      console.log(queryString);
+      const limit = req.query.limit;
+      // selects all admin boundaries ids and name_en fields where name_en field matches query string
+      knex('admin_boundaries')
+      .where('name_en', 'ILIKE', `${queryString}%`)
+      .select('id', 'name_en', 'type as level')
+      .then((results) => {
+        // serve the response. if a limit is found in the query string, limit response to that limit
+        // if not, just serve all that was found
+        limit ? res(results.slice(0, Number(limit))) : res(results);
```

Comment:
It'd be good practice to set a default limit that is also the maximum allowed and pass it in the knex query using `.limit(limit)`.

Reply:
Also you might want to check explicitly whether `limit` can be coerced to a number using `!isNaN(limit)`